src/async_client.rs

```diff
@@ -333,6 +343,191 @@ impl AsyncClient {
+            room_id_or_alias: alias.clone(),
+            third_party_signed: None,
+        };
+        self.send(request).await
+    }
+
+    /// Kick a user out of the specified room.
+    ///
+    /// Returns a `kick_user::Response`, an empty response.
+    ///
+    /// # Arguments
+    ///
+    /// * room_id - A valid `RoomId` otherwise sending will fail.
+    ///
+    /// * user_id - A valid `UserId`.
+    ///
+    /// * reason - Optional reason why the room member is being kicked out.
+    ///
+    pub async fn kick_user(
+        &mut self,
+        room_id: &RoomId,
+        user_id: &UserId,
+        reason: Option<String>,
+    ) -> Result<kick_user::Response> {
+        let request = kick_user::Request {
+            reason,
+            room_id: room_id.clone(),
+            user_id: user_id.clone(),
+        };
+        self.send(request).await
+    }
+
+    /// Leave the specified room.
+    ///
+    /// Returns a `leave_room::Response`, an empty response.
+    ///
+    /// # Arguments
+    ///
+    /// * room_id - A valid `RoomId`.
+    ///
+    pub async fn leave_room(&mut self, room_id: &RoomId) -> Result<leave_room::Response> {
+        let request = leave_room::Request {
+            room_id: room_id.clone(),
+        };
+        self.send(request).await
+    }
+
+    /// Invite the specified user by `UserId` to the given room.
+    ///
+    /// Returns a `invite_user::Response`, an empty response.
+    ///
+    /// # Arguments
+    ///
+    /// * room_id - A valid `RoomId`.
+    ///
+    /// * user_id - A valid `UserId`.
+    ///
+    pub async fn invite_user_by_id(
+        &mut self,
+        room_id: &RoomId,
+        user_id: &UserId,
+    ) -> Result<invite_user::Response> {
+        let request = invite_user::Request {
+            room_id: room_id.clone(),
+            recipient: InvitationRecipient::UserId {
+                user_id: user_id.clone(),
+            },
+        };
+        self.send(request).await
+    }
+
+    /// Invite the specified user by third party id to the given room.
+    ///
+    /// Returns a `invite_user::Response`, an empty response.
+    ///
+    /// # Arguments
+    ///
+    /// * room_id - A valid `RoomId`.
+    ///
+    /// * invite_id - A valid `UserId`.
+    ///
+    pub async fn invite_user_by_3pid(
+        &mut self,
+        room_id: &RoomId,
+        invite_id: &Invite3pid,
+    ) -> Result<invite_user::Response> {
+        let request = invite_user::Request {
+            room_id: room_id.clone(),
+            recipient: InvitationRecipient::ThirdPartyId(invite_id.clone()),
+        };
+        self.send(request).await
+    }
+
+    /// A builder to create a room and send the request.
+    ///
+    /// Returns a `create_room::Response`, an empty response.
+    ///
+    /// # Arguments
+    ///
+    /// * room - the easiest way to create this request is using the `RoomBuilder`.
+    ///
+    /// # Examples
+    /// ```ignore
```

Comment:
Examples need to at least build, run if possible. This one is doing a request so it won't be possible to run it, `no_run` can be used for this.